stix_shifter/stix_translation/src/modules/guardium/query_constructor.py

```diff
@@ -0,0 +1,524 @@
+from stix_shifter.stix_translation.src.patterns.pattern_objects import ObservationExpression, ComparisonExpression, \
+    ComparisonExpressionOperators, ComparisonComparators, Pattern, \
+    CombinedComparisonExpression, CombinedObservationExpression, ObservationOperators
+from stix_shifter.stix_translation.src.utils.transformers import TimestampToMilliseconds
+from stix_shifter.stix_translation.src.utils.transformers import TimestampToGuardium
+from stix_shifter.stix_translation.src.json_to_stix import observable
+from stix_shifter.stix_translation.src.utils.stix_pattern_parser import parse_stix
+from stix_shifter.stix_translation.src.utils import transformers
+#
+# Following added by Subroto
+import logging
+import re
+import json
+import array
+import datetime
+import re
+import sys
+import copy
+
+# Source and destination reference mapping for ip and mac addresses.
+# Change the keys to match the data source fields. The value array indicates the possible data type that can come into from field.
+#REFERENCE_DATA_TYPES = {"QUERY_FROM_DATE": ["start"],
+#                        "QUERY_TO_DATE": ["end"],"OSUser":["%"],"DBUser":"newuser",
+#                        "SHOW_ALIASES":["TRUE","FALSE"],"REMOTE_SOURCE":["%"]}
+REFERENCE_DATA_TYPES = {}
+
+#logger = logging.getLogger(__name__)
+
+class QueryStringPatternTranslator:
+    # Change comparator values to match with supported data source operators
+    comparator_lookup = {
```

Comment:
Are these the only operators supported by Guardium? An error will be thrown if a pattern is passed in with an unsupported operator.